assets/src/application/ui/display/confirm/Confirm.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useCallback, useMemo } from 'react';
+import { Button, useDisclosure } from '@chakra-ui/core';
+import { __ } from '@wordpress/i18n';
+
+import { EspressoButton } from '@application/ui/input';
+import { AlertDialog } from '@infraUI/display';
+
+import { ConfirmProps } from './types';
+
+const Confirm: React.FC<ConfirmProps> = ({ buttonProps, onConfirm, type, ...props }) => {
+	const { isOpen, onOpen, onClose } = useDisclosure();
+	const cancelRef = React.useRef();
+	const onDelete = useCallback(() => {
+		onConfirm();
+		onClose();
+	}, [onConfirm]);
+	const title = useMemo(() => {
+		if (type === 'delete') {
+			return props.title || __('Are you sure you want to delete this?');
+		}
+		return props.title;
+	}, [props.title]);
+
+	return (
+		<>
+			<EspressoButton {...buttonProps} onClick={onOpen} />
+			<AlertDialog
+				{...props}
+				cancelButton={
+					<Button ref={cancelRef} onClick={onClose}>
+						{__('No')}
+					</Button>
+				}
+				header={buttonProps?.tooltip}
+				isOpen={isOpen}
+				leastDestructiveRef={cancelRef}
+				okButton={
+					<Button variantColor='red' onClick={onDelete} ml={3}>
+						{__('Yes')}
+					</Button>
```

Comment:
was getting this error, this is why I didn't used `EspressoButton`:

![image](https://user-images.githubusercontent.com/1477453/79015969-8f5fd080-7b76-11ea-8e84-dee22d18a8a4.png)